apps/ml/receive_classes_structure.py

```diff
@@ -0,0 +1,28 @@
+from pydantic import BaseModel
+
+
+class Coordinates(BaseModel):
+    x: float
+    y: float
+    z: float
+
+
+class Accelerometer(BaseModel):
+    timestamp: date
+    data: Coordinates
+
+
+class Gyroscope(BaseModel):
+    timestamp: date
+    data: Coordinates
+
+
+class Magnetometer(BaseModel):
+    timestamp: date
+    data: Coordinates
+
+
+class ActivityTracking(BaseModel):
+    accelerometer: List[Accelerometer]
+    gyroscope: List[Gyroscope]
+    magnetometer: List[Magnetometer]
```

Comment:
Yeah this is nice but we should use it :D! This should be used to type the incoming request from ml-bridge. The `ActivityTracking` class must be extended with the `ActivityTrackingMeta` class. Basically you need to recreate this structure: 

https://github.com/damiankoper/fitly/blob/main/libs/shared/meta/src/lib/models/activity-tracking.model.ts

The overall incoming request flow is also pretty nicely visible in the ts tests made by Damian:

```ts
it('should analyse and return 10 pushups', async () => {
    // given
    const controller = app.get<AnalyzeController>(AnalyzeController);
    const date1 = DateTime.fromISO('2020-09-06T12:00:00');
    const date2 = DateTime.fromISO('2020-09-10T14:00:00');
    const interval = Interval.fromDateTimes(date1, date2);
    const meta = new ActivityTrackingMeta(interval);
    const dto = new ActivityTracking(meta);

    const responseMeta = new ActivityTrackingMeta(
      interval,
      ActivityType.PUSHUPS,
      10
    );
    MockAnalyseService.sendToML.mockResolvedValueOnce(responseMeta);

    // when
    const result = await controller.analyzeData(dto);

    // then
    expect(result).toBeInstanceOf(ActivityTrackingMeta);
    expect(result).toEqual(responseMeta);
```

So as you can see, they send to us `ActivityTracking` with all the data (we miss in our def `ActivityTrackingMeta`) and expect `ActivityTrackingMeta` in return 😄